apps/anoma_lib/lib/examples/enock_poly.ex

```diff
@@ -11,211 +11,291 @@ defmodule Examples.ENockPoly do
 
   use TypedStruct
 
-  @doc """
-  I test `Term.depth/1` and `Term.size/1` (and thus indirectly also
-  `Term.eval/3`) using a variety of terms, both closed and open.
-  """
-  @spec term_tests() :: bool()
-  def term_tests() do
-    # Closed term test cases (no variables)
-    t1 = {1, []}
-    assert Term.depth(t1) == 1
-    assert Term.size(t1) == 1
-
-    t2 = {:a, [{:a, []}, {:b, []}]}
-    assert Term.depth(t2) == 2
-    assert Term.size(t2) == 3
-
-    t3 = {:x, [{:y, [{:z, []}]}]}
-    assert Term.depth(t3) == 3
-    assert Term.size(t3) == 3
-
-    t4 = {0, [{1, []}, {2, [{3, []}, {4, []}]}, {5, []}]}
-    assert Term.depth(t4) == 3
-    assert Term.size(t4) == 6
-
-    # An example of a direct call to `Term.cata/2`, since `depth` and
-    # `size` are defined in terms of `eval` and thus do not directly
-    # test the `cata` wrapper.  This also serves as an example of
-    # providing a custom algebra to `cata` (the algebra parameter to
-    # `eval` has the same signature, so it is also an example of that).
-    # In this example we use a string constructor type and concatenate
-    # all the constructor names.
-    t8 = {"root", [{"left", []}, {"right", []}]}
-
-    algebra = fn {ctor, children} ->
-      to_string(ctor) <> Enum.join(children, "")
+  # Common tspec (constructor-check function) for poly_term tests.
+  defp common_tspec() do
+    fn
+      "zero" -> {:ok, 0}
+      "one" -> {:ok, 1}
+      "two" -> {:ok, 2}
+      _ -> {:invalid_constructor}
     end
+  end
 
-    assert Term.cata(t8, algebra) == "rootleftright"
+  # Common vspec (variable-check function) for poly_term tests.
+  defp common_vspec() do
+    fn
+      v when is_integer(v) and v >= 0 and v <= 4 -> :ok
+      _ -> {:invalid_variable}
+    end
+  end
 
-    # Open term test cases (terms with variables)
+  ####################################################################
+  ##                        TERM TESTS                              ##
+  ####################################################################
+  @doc """
+  t1: a closed term with no children.
+  """
+  def term_test_t1() do
+    res = {1, []}
+    assert Term.depth(res) == 1
+    assert Term.size(res) == 1
+    res
+  end
 
-    # A single variable (open term). By definition, its depth and size are 0.
-    v1 = "x"
-    assert Term.depth(v1) == 0
-    assert Term.size(v1) == 0
+  @doc """
+  t2: a closed term with two children.
+  """
+  def term_test_t2() do
+    res = {:a, [{:a, []}, {:b, []}]}
```

Comment:
Can we compose this example out of `term_test_t1/0`?

What do we show by doing so, what does it demonstrate about how the API works?